### PR TITLE
[INFRA & DOC] add tests for notebooks

### PR DESCRIPTION
**Pull request by Remi-Gau**
### Changes summary

- Runs the pretrained-weight notebook in CI to use as a "system test".
- Simplifies running the demo by facilitating data download from OSF

### Details

- [x] add makefile to facilitate downloading some of the data from OSF in the demo folder
- [x] adapt notebooks to run with demos in the notebook folders so users do not have to manually change the path
- [x] update notebooks to explain how the data has to be structure 
- [x] add github action workflow to test the notebook with the [nbmake pytest plugin](https://github.com/treebeardtech/nbmake)
- [x] lint notebooks content with [nb-balck](https://pypi.org/project/nb-black/)
- [x] replace path creation using string concatenation with using `os.path.join`
- [x] run on macOS
- [x] install from upstream after merging of #10

### Note 

This does not run the "full" training notebook in CI

### CODE REVIEW

File: .github/workflows/demo.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+jobs:
+
+  test_notebooks:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macOS-10.15, windows-latest]
+        python-version: ["3.9"]
```
Comment (Remi-Gau) [inline .github/workflows/demo.yml:24]: note that this only runs the demo on python 3.9.

I could switch to the latest python if #18 is solved

File: .github/workflows/demo.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      - name: Clone repo
+        uses: actions/checkout@v3
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install deepmreye from github
+        run: pip install git+https://github.com/Remi-Gau/DeepMReye@setup
```
Comment (Remi-Gau) [inline .github/workflows/demo.yml:44]: Note that this way of installing tries to mimic how a user would install the demo.

Because the masks are not downloaded if installed from the upstream repo (see #6 and this [comment PR10](https://github.com/DeepMReye/DeepMReye/pull/10#issuecomment-1176266742)), I am currently installing from a fork on my branch where this problem has been fixed.

File: notebooks/deepmreye_example_usage.ipynb [hunk 3]
```diff
@@ -9531,7 +9531,49 @@
     "<a name=\"download\"></a>\n",
     "## Download exemplary data\n",
     "\n",
-    "To run this notebook, you will need to download a sample dataset from Open Science Framework: [https://osf.io/mrhk9/](https://osf.io/mrhk9/). The dataset comprises fMRI data of 12 exemplary participants, who fixated at various locations on the screen. These locations are hard coded in the present script, but can be easily replaced e.g. by camera eye tracking labels.  "
+    "To run this notebook, you will need to download a sample dataset from Open Science Framework: [https://osf.io/mrhk9/](https://osf.io/mrhk9/). \n",
```
Comment (Remi-Gau) [inline notebooks/deepmreye_example_usage.ipynb:9534]: To better view the notebook:

https://github.com/Remi-Gau/DeepMReye/blob/test_demo/notebooks/deepmreye_example_usage.ipynb

File: notebooks/deepmreye_example_usage_pretrained_weights.ipynb [hunk 4]
```diff
@@ -9525,9 +9525,54 @@
    "source": [
     "---\n",
     "<a name=\"download\"></a>\n",
-    "## Download exemplary data\n",
+    "## Download exemplary data and pretrained model weights\n",
```
Comment (Remi-Gau) [inline notebooks/deepmreye_example_usage_pretrained_weights.ipynb:9528]: To better view this notebook: https://github.com/Remi-Gau/DeepMReye/blob/test_demo/notebooks/deepmreye_example_usage_pretrained_weights.ipynb

**Comment by Remi-Gau** [comment]
Test in CI seems brittle: ran fine on my branch [here](https://github.com/Remi-Gau/DeepMReye/actions/runs/2631761204)

Will try to make it less so: expect a few "noisy commits" while I try things.

**Comment by Remi-Gau** [comment]
To make my life simple I decided to only test on ubuntu for now. :see_no_evil:

**Comment by CYHSM** [comment]
Cool thats very helpful, so I guess we first merge #10 then we can adjust the paths in here and then merge this. Any chance we can test this across all operating systems?

**Comment by Remi-Gau** [comment]
> Cool thats very helpful, so I guess we first merge #10 then we can adjust the paths in here

This is done.

> Any chance we can test this across all operating systems?

For this I would have to use something else than make to get the data and the model.

Maybe a set of small python scripts would be better and be usable for all OS. Maybe better as a separate PR, no?

**Comment by CYHSM** [comment]
Ok so it seems to run on Ubuntu and macOS, should be okay for most users. I guess if the tests pass, the notebook should also work on windows, just with manual scripting before. Could add that as a separate issue, we would need another makefile for windows, right?

**Comment by Remi-Gau** [comment]
> we would need another makefile for windows, right?

make is not really a windows thing

> The pipfile tensorflow version could also change, no?

maybe even better remove it completely, it is a duplicate of other information: most likely will confuse users

> Also not sure if demo.yaml is the best name for the file

any suggestions ?
